### PR TITLE
Remove `StorageClient` from `azure_storage_queues`

### PR DESCRIPTION
**Pull request by rylev**
This is the next step in the process of removing `StorageClient` following #1062 and #1057.

Creating the various clients in `azure_storage_queues` now mirrors how things work with blob storage, data lake, and cosmos.

### CODE REVIEW

File: sdk/storage_queues/src/clients/pop_receipt_client.rs [hunk 1]
```diff
@@ -41,19 +40,27 @@ impl PopReceiptClient {
...
-                .iter()
-                .map(std::ops::Deref::deref),
-        )?;
+    pub(crate) fn url(&self) -> azure_core::Result<url::Url> {
+        let url = format!(
+            "{}/{}",
+            self.client.messages_url()?,
+            self.pop_receipt.message_id()
+        );
+        let mut url = url::Url::parse(&url)?;
```
Comment (gorzell) [inline sdk/storage_queues/src/clients/pop_receipt_client.rs:63]: **Suggestion:** I think this is a bit cleaner:
```rust
pub(crate) fn url(&self) -> azure_core::Result<url::Url> {
        let mut url =self.client.messages_url()?.join(self.pop_receipt.message_id())?;
        url.query_pairs_mut()
            .append_pair("popreceipt", self.pop_receipt.pop_receipt());
        Ok(url)
    }
```

**Review by gorzell — CHANGES_REQUESTED** [review]
Maybe not something to address in this PR, but I am not convinced that all of these `url` construction functions should actually be fallible. In many cases it seems like if they can't actually construct a valid URL they should actually `panic`. In most cases they aren't really using information provided by the caller anyway, so I am not really sure how you could even recover from the error.

**Review by bmc-msft — APPROVED** [review]

File: sdk/storage_queues/src/clients/pop_receipt_client.rs [hunk 2]
```diff
@@ -41,19 +40,27 @@ impl PopReceiptClient {
...
+    pub(crate) fn finalize_request(
+        &self,
+        url: url::Url,
+        method: azure_core::Method,
+        headers: azure_core::headers::Headers,
+        request_body: Option<azure_core::Body>,
+    ) -> azure_core::Result<Request> {
+        self.client
+            .finalize_request(url, method, headers, request_body)
     }
```
[the remark this answers is not in the document]
Comment (rylev) [reply sdk/storage_queues/src/clients/pop_receipt_client.rs:55]: Yea, this probably makes sense.

**Comment by rylev** [comment]
@gorzell the urls use the account name which is user provided - we could check at construction that the username is url safe and then not have to return fallible url methods, but I would leave that for a follow-up.

File: sdk/storage_queues/src/clients/pop_receipt_client.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rylev) [reply sdk/storage_queues/src/clients/pop_receipt_client.rs:55]: On second thought I'd like to do this in a separate PR.

**Review by gorzell — APPROVED** [review]

**Pull request merged**